### ISSUE TITLE
New spell idea: Magical barrier and Physical barrier

### ISSUE THREAD

**Issue by Howdidibreakit** [00-Evan/shattered-pixel-dungeon#1323]
The magical barrier looks like magical flames, and will reflect any spell back at the caster if it passes through it, and also applies crippled to anything that goes through. It takes a scroll of anti-magic and an arcane catalyst. The physical barrier looks like there's about to be rockfall, and creatures that go through are paralyzed and take 2-5*(depth/2) damage every turn until it ends. Missile weapons are stopped when they pass through, except for the huntress's bow. They both last 10 turns, but can be canceled by throwing a potion of purity at it.

**Comment by Howdidibreakit** [comment]
I forgot to mention, the physical barrier takes a potion of earthen armor and an arcane catalyst. They both cost 8 energy to make.

**Comment by 00-Evan** [comment]
This sounds like it fulfills pretty much the same function that items like the stone of flock, wolly bomb, or stone of deep sleep already do. There's no need to add additional ways to block out enemies that are more circumstantial. Enemies would also have to either know to avoid these barriers, or they would be extremely overpowered (e.g. a magic enemy would just kill themselves on it, or ignore it entirely, a melee enemy would be a free kill if it walked into the barrier, so it would have to avoid it).

**Comment by SujayDalai02** [comment]
@00-Evan well, it might be okay if those scrolls(or maybe spells? idk) completely immune a number of hits of the corresponding type made against player?

(Sorry for a weird suggestion)

**Comment by Howdidibreakit** [comment]
Yeah, maybe they could like just trigger once, like a trap or plant effect, but given that I could say "like" it's something, then it's not actually useful.
